main.go

```diff
@@ -491,6 +491,48 @@ func main() {
 			}
 		})
 
+		http.HandleFunc("/get-url", func(w http.ResponseWriter, r *http.Request) {
+			queryParameters := r.URL.Query()
+			var targetUrl string
+			if queryParameters.Has("url") {
+				targetUrl = queryParameters.Get("url")
+			} else {
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
+
+			resp, err := http.Get(targetUrl)
+			if err != nil {
+				http.Error(w, fmt.Sprintf("Request to target failed: %v", err), http.StatusBadGateway)
+				return
+			}
+			defer resp.Body.Close()
+
+			if r.Method == http.MethodHead {
+				// Just return the status code
+				w.WriteHeader(resp.StatusCode)
+				return
+			}
+
+			if r.Method == http.MethodGet {
+				// Copy headers
+				for key, values := range resp.Header {
+					for _, v := range values {
+						w.Header().Add(key, v)
+					}
+				}
+
+				// Set status code and copy body
+				w.WriteHeader(resp.StatusCode)
+				_, _ = io.Copy(w, resp.Body)
+				return
+			}
```

Comment:
I know we have only test code here, but HTTP spec wise, a HEAD request usually returns the same status code and headers as if the same request would have been performed as GET request - just without the body. If it does not hurt, I'd go like this:

Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Methods/HEAD

```suggestion
			// Copy headers
			for key, values := range resp.Header {
				for _, v := range values {
					w.Header().Add(key, v)
				}
			}

			// Set status code
			w.WriteHeader(resp.StatusCode)

			if r.Method == http.MethodGet {
				_, _ = io.Copy(w, resp.Body)
				return
			}
```